blocks/library/latest-posts/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import Placeholder from 'components/placeholder';
+
+/**
+ * Internal dependencies
+ */
+import { registerBlockType } from '../../api';
+import { getLatestPosts } from './data.js';
+
+registerBlockType( 'core/latestposts', {
+	title: wp.i18n.__( 'Latest Posts' ),
+
+	icon: 'list-view',
+
+	category: 'rest-api',
+
+	defaultAttributes: {
+		poststoshow: 5,
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+
+			this.state = {
+				latestPosts: [],
+			};
+
+			const { poststoshow } = this.props.attributes;
+
+			getLatestPosts( poststoshow )
+				.then( latestPosts => this.setState( { latestPosts } ) );
+		}
+
+		renderPostsLoading() {
```

Comment:
And in the case they're not simple, it's a good sign the component could do to be broken up into multiple smaller self-contained components.